pulumitest/opttest/opttest.go

```diff
@@ -109,16 +109,54 @@ type Options struct {
 	ExtraWorkspaceOptions []auto.LocalWorkspaceOption
 }
 
+// Copy creates a deep copy of the current options.
+func (o *Options) Copy() *Options {
```

Comment:
`github.com/pulumi/pulumi/sdk/v3/go/common/util/deepcopy` could probably replace this code, while guarding against forgetting to add future options to the Copy method.

Reply:
Yup, this makes sense.